roofit/hs3/src/RooJSONFactoryWSTool.cxx

```diff
@@ -495,8 +495,8 @@ std::vector<std::vector<int>> RooJSONFactoryWSTool::generateBinIndices(const Roo
    std::vector<std::vector<int>> combinations;
    std::vector<int> vars_numbins;
    vars_numbins.reserve(vars.size());
-   for (auto &absv : vars) {
-      vars_numbins.push_back(((RooRealVar *)absv)->numBins());
+   for (const auto &absv : static_range_cast<RooRealVar *>(vars)) {
```

Comment:
```suggestion
   for (const auto *absv : static_range_cast<RooRealVar *>(vars)) {
```
Some compilers in the CI complain about references to a temporary pointer here. If you use the `range_cast` with a pointer type, you also need to copy that pointer with `const auto*`.